controllers/prep_handlers.go

```diff
@@ -481,6 +450,51 @@ func (r *ImageBasedUpgradeReconciler) handlePrep(ctx context.Context, ibu *lcav1
 	return
 }
 
+// updatePrepTaskForPrecacheJob check up on precache job and update PrepTask as needed
+func (r *ImageBasedUpgradeReconciler) updatePrepTaskForPrecacheJob(ctx context.Context) {
+	if _, err := precache.GetJob(ctx, r.Precache.Client); err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.Log.Info("Precache job not launched yet")
+		} else {
+			r.failPrepTask(err, "prep stage failed, unable to get precache job")
+		}
+		return
+	}
+
```

Comment:
This should be handled in the following precache job status check already as it needs to get the job first. It seems like a duplicate?

Reply:
Yeah like the getJob call was buried deep and with ton of error wraps...wanted to keep it as it is. But yeah makes sense to just refactor!

Reply:
Removed the extra call and instead bubbling up the error